lib/splitclient-rb/clients/split_client.rb

```diff
@@ -1,114 +1,131 @@
-    @config = config
-
-    @splits_repository = splits_repository
-    @segments_repository = segments_repository
-    @impressions_repository = impressions_repository
-    @metrics_repository = metrics_repository
-
-    @adapter = adapter
-  end
+    #
+    # Creates a new split client instance that connects to split.io API.
+    #
+    # @param api_key [String] the API key for your split account
+    #
+    # @return [SplitIoClient] split.io client instance
+    def initialize(api_key, config = {}, adapter = nil, splits_repository, segments_repository, impressions_repository, metrics_repository)
+      @config = config
+
+      @splits_repository = splits_repository
+      @segments_repository = segments_repository
+      @impressions_repository = impressions_repository
+      @metrics_repository = metrics_repository
+
+      @adapter = adapter
+    end
+
+    def get_treatments(key, split_names, attributes = nil)
+      bucketing_key, matching_key = keys_from_key(key)
+      bucketing_key = matching_key if bucketing_key.nil?
 
-  def get_treatments(key, split_names, attributes = nil)
-    bucketing_key, matching_key = keys_from_key(key)
-    bucketing_key = matching_key if bucketing_key.nil?
+      treatments_with_labels =
+        @splits_repository.get_splits(split_names).each_with_object({}) do |(name, data), memo|
+          memo.merge!(name => get_treatment(key, name, attributes, data, false, true))
+        end
 
-    treatments =
-      @splits_repository.get_splits(split_names).each_with_object({}) do |(name, data), memo|
-        memo.merge!(name => get_treatment(key, name, attributes, data, false))
+      if @config.impressions_queue_size > 0
+        @impressions_repository.add_bulk(matching_key, bucketing_key, treatments_with_labels, (Time.now.to_f * 1000.0).to_i)
       end
 
-    if @config.impressions_queue_size > 0
-      @impressions_repository.add_bulk(matching_key, bucketing_key, treatments, (Time.now.to_f * 1000.0).to_i)
+      split_names = treatments_with_labels.keys
+      treatments = treatments_with_labels.values.map { |v| v[:treatment] }
+
+      Hash[split_names.zip(treatments)]
     end
 
-    treatments
-  end
+    #
+    # obtains the treatment for a given feature
+    #
+    # @param key [String/Hash] user id or hash with matching_key/bucketing_key
+    # @param split_name [String/Array] name of the feature that is being validated or array of them
+    #
+    # @return [String/Hash] Treatment as String or Hash of treatments in case of array of features
+    def get_treatment(key, split_name, attributes = nil, split_data = nil, store_impressions = true, multiple = false)
+      bucketing_key, matching_key = keys_from_key(key)
+      bucketing_key = matching_key if bucketing_key.nil?
+
+      if matching_key.nil?
+        @config.logger.warn('matching_key was null for split_name: ' + split_name.to_s)
+        return parsed_treatment(multiple, { label: 'matching_key was null', treatment: Treatments::CONTROL })
```

Comment:
This is not part of the spec. Make is `exception` for now which is part of the spec..